databricks/sdk/service/_internal.py

```diff
@@ -10,7 +10,7 @@ def _from_dict(d: Dict[str, any], field: str, cls: Type) -> any:
 
 def _repeated(d: Dict[str, any], field: str, cls: Type) -> any:
     if field not in d or not d[field]:
-        return None
+        return []
```

Comment:
The main thing I'd review is how Databricks proto deserialization handles empty lists and if there are cases where APIs POST APIs don't support an empty array. For example, the `tasks` field in jobs comes to mind. In the case of Jobs it's fine to pass [] — we don't actually allow jobs with zero tasks but ignore this array if there is a task in the settings body of the job. Are there other cases where `[]` would break

Reply:
@lennartkats-db this is about deserialization in the SDK layer - of what Databricks platform has returned. it has nothing to do with how SDK requests are serialized.

specifically, this makes response deserialization to behave more similar to how proto deserializes

Reply:
From what you told me this influences if you get a job without a task property back from the SDK. Or a job with an empty array as the tasks property.

The challenge is that if we return an empty array, then it needs to be possible to submit that same JSON blob back to the API with e.g. the jobs create API. Otherwise users get very surprising behavior.

Reply:
To clarify: this isn't about the JSON blob, it's about the materialized job object that is returned. The equivalent of what you're asking about in the SDK is whether someone can do
```py
j = w.jobs.get(job_id=123)
w.jobs.create(**j.settings)
```
In my understanding, the protobuf library does not distinguish between missing and empty list for repeated fields. If an API needs to be able to distinguish between these cases (e.g. an Update API where one of the fields is a list that can be optionally updated), a field mask or other mechanism is necessary to explicitly inform the server whether this is needed or not.

Reply:
@mgyucht 
> j = w.jobs.get(job_id=123)
> w.jobs.create(**j.settings)

👍 We're talking about the same thing here.

So I mainly just want confirmation on
> The main thing I'd review is how Databricks proto deserialization handles empty lists and if there are cases where APIs POST APIs don't support an empty array.

It sounds like in your understanding, proto deserialization addresses this issue. I suspect that you're right. But could you 100% confirm that understanding?

Reply:
I'll double-check with @alexc-db, he probably would be the best to confirm this.